### DISCUSSION TITLE
How to build Xamarin Forms project with UWP on Macos?

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3403] [Extension Q&A] [answered]
I have a Xamarin forms project that has UWP project in it (iOS, Android, and UWP)

I created a configuration that excludes UWP from the build. I did this to avoid building UWP, since it fails on MacOS.

But, when I run "NuGetRestore(BuildParameters.SolutionFile.ToString());"

Which does a restore on the solution which has about 10 different projects, how can I exclude the project "myproject.UWP.csproj"?

Is there a way, via a cake extension, that I can remove the project from the build?

**Comment by devlead** [comment, marked as answer]
UWP will only build and work on Windows. 

For cross platform UWP you could take a look at UNO
https://platform.uno/

To exclude I would have a build configuration that only builds subset of projects.

Some docs about that here
https://docs.microsoft.com/en-us/visualstudio/mac/create-and-edit-configurations?view=vsmac-2019

**Reply by nick5454** [reply]
@devlead Well the issue is that I don't want to build UWP, just the IOS project. I changed the configuration to not include the project. That works, but the restore is trying to restore it.

What is happening is that when I restore the solution, it tries to restore on the UWP project. It seems that NugetRestore ignores the configuration. I was hoping someone knew the best way to either remove or tell Nuget to exclude the UWP project when restoring.

I guess I could restore each project that doesn't have ".UWP" in the project name

I need it in there because UWP is the fastest development and we ignore the project when deploying to the app center.

**Reply by devlead** [reply]
By having build configurations you can pass that as an parameter to msbuild.

You can choose which projects build for which configurations.

You can restore using MSBuild by passing restore target, which should honor solution build config.

https://docs.microsoft.com/en-us/nuget/consume-packages/package-restore#restore-using-msbuild

**Reply by nick5454** [reply]
@devlead There is an issue with the MSBuild on Mac. If it has a project with the old Nuget, 2.7 I think the project will never find the packages, because they are not downloaded. You have to migrate the package.config to PackageReference. This stumped me for hours.
